cdap-data-fabric/src/main/java/co/cask/cdap/spi/data/table/field/Field.java

```diff
@@ -64,8 +66,13 @@ public boolean equals(Object o) {
       return false;
     }
     Field field = (Field) o;
-    return Objects.equals(fieldType, field.fieldType) &&
-      Objects.equals(value, field.value);
+    if (!Objects.equals(fieldType, field.fieldType)) {
+      return false;
+    }
+    if (fieldType.getType() == FieldType.Type.BYTES) {
+      return Bytes.equals((byte[]) value, (byte[]) field.getValue());
+    }
+    return Objects.equals(value, field.value);
```

Comment:
please change hashCode(), too.

Reply:
Also should toString() handle bytes array?